### PR TITLE
Add national party schedule a/b endpoints 

### LINKED ISSUE #5786: Build national party account endpoints [Transaction level]
### What we’re after

After the [MV](https://github.com/fecgov/openFEC/pull/5780) is merged, we need to build out the endpoints for the national party account. 

### Designs

**Disbursements datatable updated mockup with amount and data as default sortable columns:**

Screenshot 2024-04-24 at 2 07 19 PM

**Receipts datatable updated mockup with amount and data as default sortable columns:**

Screenshot 2024-04-24 at 2 08 14 PM

These should be two endpoints under a new top level national party endpoint  for the transaction level data
national_party/schedule_a
national_party/schedule_b

### Related ticket(s)

- Receipts datatable:  https://github.com/fecgov/fec-cms/issues/6179
- Disbursements datatable: https://github.com/fecgov/fec-cms/issues/6180

### Completion criteria

- [ ] National Party accounts receipt/disbursements endpoints are built and supports the design above

### PR DESCRIPTION
**Pull request by pkfec**
## Summary 

Create two new national party account endpoints (Transaction level) under swagger docs

1. `/national_party/schedule_a/`
2. `/national_party/schedule_b/`

### Required reviewers
1 frontend and 2 or more backend developers

## Screenshots
Screen Shot 2024-04-24 at 11 38 58 PM

## How to test
On local:

- run `git checkout feature/nation-party-endpoints`
- run `pyenv activate venv-api`
- run `pip install -r requirements.txt` 
- run `pip install -r requirements-dev.txt` 
- run `pytest`
- run `flask run`
- test filters under national_party/schedule_a endpoint:
 http://127.0.0.1:5000/v1/national_party/schedule_a/?page=1&per_page=20&sort=-contribution_receipt_date&sort_hide_null=false&sort_null_only=false

- test filters under national_party/schedule_b endpoint: 
http://127.0.0.1:5000/v1/national_party/schedule_b/?page=1&per_page=20&sort=-disbursement_date&sort_hide_null=false&sort_null_only=false

**OR**

Deploy a test branch to Dev space and test the NPA filters

### CODE REVIEW

**Review by cnlucas — COMMENTED** [review]
Can we remove these NULL fields. I think ideally they wouldn't be in the MV, but can we at least take them out of the model? 
candidate_first_name
candidate_id
candidate_last_name
candidate_middle_name
candidate_name
candidate_office
candidate_office_district
candidate_office_full
candidate_office_state
candidate_office_state_full
conduit_committee_city
conduit_committee_id	
conduit_committee_name	
conduit_committee_state
conduit_committee_street1	
conduit_committee_street2
conduit_committee_zip
candidate_prefix
candidate_suffix

File: webservices/resources/national_party.py
Comment (cnlucas) [inline webservices/resources/national_party.py:1]: @JonellaCulmer 's design requires sort for recipient, source, account, date, and amount. This PR only permits sorting by date and amount. Was there a change to the design @pkfec? I know you all have been in meetings about the design--but I'm just comparing to the old ticket designs.

Comment (cnlucas) [reply webservices/resources/national_party.py:1]: Min/max date filters are not working for me.

Comment (pkfec) [reply webservices/resources/national_party.py:1]: @cnlucas After discussing with @JonellaCulmer @patphongs and @hcaofec the default sort is prioritize to be amount and data columns for national party account endpoints. The latest mockups are available in the frontend tickets here in https://github.com/fecgov/fec-cms/issues/6179 && https://github.com/fecgov/fec-cms/issues/6180

**Comment by pkfec** [comment]
> Can we remove these null fields. I think ideally they wouldn't be in the MV, but can we at least take them out of the model? Can someone confirm what I'm seeing--that they all are null? organization_type organization_type_full candidate_first_name candidate_id candidate_last_name candidate_middle_name candidate_name candidate_office candidate_office_district candidate_office_full candidate_office_state candidate_office_state_full conduit_committee_city conduit_committee_id conduit_committee_name conduit_committee_state conduit_committee_street1 conduit_committee_street2 conduit_committee_zip candidate_prefix candidate_suffix

Sure, I can take a look and confirm if all the fields are null.

**Comment by pkfec** [comment]
> conduit_committee_city
> conduit_committee_id
> conduit_committee_name
> conduit_committee_state
> conduit_committee_street1
> conduit_committee_street2
> conduit_committee_zip

 @PaulClark2 suggested removing the organization and conduit* fields since party committees typically lack that data. However, he recommended keeping the candidate information since some party committees do have relevant data for it.
 cc @hcaofec

**Comment by pkfec** [comment]
> > conduit_committee_city
> > conduit_committee_id
> > conduit_committee_name
> > conduit_committee_state
> > conduit_committee_street1
> > conduit_committee_street2
> > conduit_committee_zip
> 
> @PaulClark2 suggested removing the organization and conduit* fields since party committees typically lack that data. However, he recommended keeping the candidate information since some party committees do have relevant data for it. cc @hcaofec

I removed the organization and conduit* fields from the NPA model file

**Review by fec-jli — APPROVED** [review]
It works as expected.
Could we consider creating a separate ticket to add a 'Zipcode' validation function in args.py? This would help eliminate the excessive duplication of Zipcode validation code in many places.

**Comment by pkfec** [comment]
> It works as expected. Could we consider creating a separate ticket to add a 'Zipcode' validation function in args.py? This would help eliminate the excessive duplication of Zipcode validation code in many places.

Thanks for the feedback @fec-jli. I can make up a ticket to move the Zipcode validation function to webservies.utils.py instead of args.py as most validation happens in utils.py

**Review by tmpayton — APPROVED** [review]
Really awesome job @pkfec!

**Review by hcaofec — APPROVED** [review]
Great work!

**Review by johnnyporkchops — APPROVED** [review]
Endpoints look good . Great job Priya!

**Comment by pkfec** [comment]
I added `recipient_zip` as a filter for sb endpoint.

**Comment by JonellaCulmer** [comment]
@pkfec Just two things for the endpoints:
- Change receipt_type and disbursement_type filter descriptions to say: "National party account receipt types:" or "National party account disbursement types:"
- I don't see city or ZIP for the recipient on receipts endpoint, and for the spender on the disbursements endpoint.

**Comment by pkfec** [comment]
> @pkfec Just one thing for the endpoints:
> 
> * Change receipt_type and disbursement_type filter descriptions to say: "National party account receipt types:" or "National party account disbursement types:"

@JonellaCulmer I updated ^^ filter descriptions in docs.py

**Review by JonellaCulmer — APPROVED** [review]
This looks really great, @pkfec. Thanks for all your hard work on this.

**Pull request merged**